cmd/admin.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 The Nakama Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/gorhill/cronexpr"
+	"github.com/satori/go.uuid"
+	"github.com/uber-go/zap"
+	"net/url"
+	"os"
+)
+
+type adminService struct {
+	DSNS   string
+	logger zap.Logger
+}
+
+func AdminParse(args []string, logger zap.Logger) {
+	if len(args) == 0 {
+		logger.Fatal("Admin requires a subcommand. Available commands are: 'create-leaderboard'.")
+	}
+
+	var exec func([]string, zap.Logger)
+	switch args[0] {
+	case "create-leaderboard":
+		exec = createLeaderboard
+	default:
+		logger.Fatal("Unrecognized admin subcommand. Available commands are: 'create-leaderboard'.")
+	}
+
+	exec(args[1:], logger)
+	os.Exit(0)
+}
+
+func createLeaderboard(args []string, logger zap.Logger) {
+	var dsns string
+	var id string
+	var authoritative bool
+	var sortOrder string
+	var resetSchedule string
+	var metadata string
+
+	flags := flag.NewFlagSet("admin", flag.ExitOnError)
+	flags.StringVar(&dsns, "db", "root@localhost:26257", "CockroachDB JDBC connection details.")
+	flags.StringVar(&id, "id", "", "ID to assign to the leaderboard.")
+	flags.BoolVar(&authoritative, "authoritative", false, "True if clients may not submit scores directly, false otherwise.")
+	flags.StringVar(&sortOrder, "sort", "descending", "Leaderboard sort order, 'asc' or 'desc'.")
+	flags.StringVar(&resetSchedule, "reset", "", "Optional reset schedule in CRON format.")
+	flags.StringVar(&metadata, "metadata", "{}", "Optional additional metadata as a JSON string.")
+
+	if err := flags.Parse(args); err != nil {
+		logger.Fatal("Could not parse admin flags.")
+	}
+
+	if dsns == "" {
+		logger.Fatal("Database connection details are required.")
+	}
+
+	query := `INSERT INTO leaderboard (id, authoritative, sort_order, count, reset_schedule, metadata)
+	VALUES ($1, $2, $3, 0, $4, $5)`
+	params := []interface{}{}
+
+	// ID.
+	if id == "" {
+		params = append(params, uuid.NewV4().Bytes())
+	} else {
+		params = append(params, []byte(id))
+	}
+
+	// Authoritative.
+	params = append(params, authoritative)
+
+	// Sort order.
+	if sortOrder == "asc" {
+		params = append(params, 0)
+	} else if sortOrder == "desc" {
+		params = append(params, 1)
+	} else {
+		logger.Fatal("Invalid sort value, must be 'asc' or 'desc'.")
+	}
+
+	// Count is hardcoded in the INSERT above.
```

Comment:
We could probably strip count from the query above. It has `0` as the default value for the field.